### PR TITLE
Fixes app crash when reopening iPad app after exiting it with a modal module visible.

### LINKED ISSUE #699: iPad MITModulePresentationModal types crash
Close the app with a MITModulePresentationModal type open (Settings, Links, About).
Open the app and select a MITModulePresentationModal type. 
Click on background to close MITModulePresentationModal and it will crash.

### PR DESCRIPTION
**Pull request by bskinner**
Fixes #699.

### CODE REVIEW

File: App Delegate/MITSlidingViewController.m [hunk 1]
```diff
@@ -448,16 +453,15 @@ - (void)_showModuleWithModuleItem:(MITModuleItem*)moduleItem animated:(BOOL)anim
...
-    // (bskinner - 2014.11.07, updated 2014.12.01)
     UIViewController *moduleViewController = [self _moduleViewControllerWithName:moduleItem.name];
     NSAssert(moduleViewController,@"module with name %@ does not have an associated view controller.",moduleItem.name);
 
-
-    [[MIT_MobileAppDelegate applicationDelegate] showModuleWithTag:moduleItem.name animated:animated];
+    [self _willHideTopViewController:self.visibleViewController];
+    __weak MITSlidingViewController *weakSelf = self;
+    [self setVisibleViewController:moduleViewController animated:animated completion:^{
+        MITSlidingViewController *blockSelf = weakSelf;
```
Comment (Smartcop) [inline App Delegate/MITSlidingViewController.m]: should you check if you have a blockSelf first?  if (blockSelf) { }

File: App Delegate/MITSlidingViewController.m [hunk 2]
```diff
@@ -448,16 +453,15 @@ - (void)_showModuleWithModuleItem:(MITModuleItem*)moduleItem animated:(BOOL)anim
...
+        [blockSelf _didShowTopViewController:blockSelf.visibleViewController];
```
Comment (Smartcop) [inline App Delegate/MITSlidingViewController.m]: should you check if you have a blockSelf first?  if (blockSelf) { }

**Comment by Smartcop** [comment]
I added a note
@bskinner

File: App Delegate/MITSlidingViewController.m (hunk 2 above)
Comment (bskinner) [inline App Delegate/MITSlidingViewController.m]: No need. If _blockSelf_ is nil here, the entire method call is a NOOP and there's no other operations being performed.

**Pull request merged**